src/main/java/org/candlepin/subscriptions/tally/MetricUsageCollector.java

```diff
@@ -24,261 +24,140 @@
-    Additionally, it assesses whether the first event timestamp is after the host last seen.
-    If this condition holds false, then the effectiveStartDateTime is adjusted from the first day of the month to
-    the end of current hour. Otherwise, the start date is same as the beginning of the user start range,
-    it extends until the specified passed end date.
-     */
-    if (maxLastSeenTimestamp.isEmpty()
-        || !firstEventTimestamp.isAfter(maxLastSeenTimestamp.get())) {
-      int eventLastMonth =
-          firstEventTimestamp.getMonth().getValue() - OffsetDateTime.now().getMonth().getValue();
-      if (eventLastMonth < 0) {
-        effectiveStartDateTime = clock.startOfMonth(firstEventTimestamp);
-      } else {
-        effectiveStartDateTime = clock.startOfMonth(range.getStartDate());
-      }
-      effectiveEndDateTime = range.getEndDate();
-      log.info(
-          "We appear to be retallying; adjusting start and end from [{} : {}] to [{} : {}]",
-          range.getStartString(),
-          range.getEndString(),
-          effectiveStartDateTime,
-          effectiveEndDateTime);
-      isRecalculating = true;
-    } else {
-      effectiveStartDateTime = firstEventTimestamp; // Earliest timestamp for an unprocessed event
-      effectiveEndDateTime = range.getEndDate(); // now - buffer
-      log.info(
-          "New tally! Adjusting start and end from [{} : {}] to [{} : {}]",
-          range.getStartString(),
-          range.getEndString(),
-          effectiveStartDateTime,
-          effectiveEndDateTime);
-      isRecalculating = false;
+    if (!accountServiceInventoryRepository.existsById(inventoryId)) {
+      accountServiceInventoryRepository.save(new AccountServiceInventory(inventoryId));
     }
 
-    if (isRecalculating) {
-      log.info("Clearing monthly totals for {} instances", existingInstances.size());
-      existingInstances
-          .values()
-          .forEach(
-              instance ->
-                  instance.clearMonthlyTotals(effectiveStartDateTime, effectiveEndDateTime));
-    }
-
-    Map<OffsetDateTime, AccountUsageCalculation> accountCalcs =
-        collectHourlyCalculations(
-            accountServiceInventory, effectiveStartDateTime, effectiveEndDateTime);
-    accountServiceInventoryRepository.save(accountServiceInventory);
+    var hostsByInstanceId =
+        hostRepository
+            .findAllByOrgIdAndInstanceIdIn(
+                orgId, events.stream().map(Event::getInstanceId).collect(Collectors.toSet()))
+            .collect(Collectors.toMap(Host::getInstanceId, Function.identity()));
+
+    for (Event event : events) {
+      Host host = hostsByInstanceId.getOrDefault(event.getInstanceId(), new Host());
+
+      // Determine if the Event has already been applied to this host and skip it if it is.
+      // This is important in the case that we attempt to apply the same Event multiple times.
```

Comment:
Some additional context that would be nice to have in this comment:
1. An example of when we might attempt to apply the same event.
2. What happens otherwise.

Reply:
DONE